articles/dms/howto-mysql-to-azure-mysql-powershell.md

```diff
@@ -0,0 +1,657 @@
+
+        # Check if this table is on the selected list
+        foreach ($nextSelectedTable in $SelectedTables) {
+            $isSelected = $tgtTable.Name -ieq $nextSelectedTable;
+
+            if ($isSelected) {
+                break;
+            }
+        }
+
+        if (-not $isSelected) {
+            continue;
+        }
+    }
+
+    $tableMap.Add("$($SourceDatabaseName).$tableName", "$($TargetDatabaseName).$tableName");
+}
+
+if ($tableMap.Count -le 0) { throw "ERROR: Could not create table mapping." }
+LogMessage -Message "Migration table mapping created for $($tableMap.Count) tables."
+```
+
+## Step 9: Create and configure the migration task
+
+After building the table mapping, you will create the the migration task of type *Migrate.MySql.AzureDbForMySql* and configure the properties.
+
+The following script creates the migration task and sets the connections, database names and table mapping.
+
+```powershell
+# Create and configure the migration scenario based on the connections
+# and the table mapping
+$offlineMigTaskProperties = @{
+    "input"    = @{
+        "sourceConnectionInfo"  = $null;
+        "targetConnectionInfo"  = $null;
+        "selectedDatabases"     = $null;
+        "optionalAgentSettings" = @{
+            "EnableCacheBatchesInMemory"         = $true;
+            "DisableIncrementalRowStatusUpdates" = $true;
+        };
+        "startedOn"             = $null;
+    };
+    "taskType" = "Migrate.MySql.AzureDbForMySql";
+};
+$offlineSelectedDatabase = @{
+    "name"               = $null;
+    "targetDatabaseName" = $null;
+    "tableMap"           = $null;
+};
+
+LogMessage -Message "Preparing migration scenario configuration ..." -IsProcessing $true
+
+# Select the database to be migrated
+$offlineSelectedDatabase.name = $SourceDatabaseName;
+$offlineSelectedDatabase.tableMap = New-Object PSObject -Property $tableMap;
+$offlineSelectedDatabase.targetDatabaseName = $TargetDatabaseName;
+
+# Set connection info and the database mapping
+$offlineMigTaskProperties.input.sourceConnectionInfo = $sourceConnInfo;
+$offlineMigTaskProperties.input.targetConnectionInfo = $targetConnInfo;
+$offlineMigTaskProperties.input.selectedDatabases = @($offlineSelectedDatabase);
+$offlineMigTaskProperties.input.startedOn = [System.DateTimeOffset]::UtcNow.ToString("O");
+```
+
+## Step 10: Configure performance tuning parameters
+
+As pert of the PowerShell module, there are few optional parameters available, which can be tuned based on the environment. These parameters can be used to improve the performance of the migration task. All these parameters are optional and their default value is NULL.
+
+> [!NOTE]
+> The following performance configurations have shown increased throughput during migration on Premium sku.
+> * DesiredRangesCount = 4
+> * QueryTableDataRangeTaskCount = 8
+> * WriteDataRangeBatchTaskCount = 12
+> * DelayProgressUpdatesInStorageInterval = 30 seconds
+> * MaxBatchSize = 4 MB
```

Comment:
Default of MaxBatchSize is 4 MB today, so we probably don't have to mention it needs to be customized?

Reply:
Done